File: src/main/java/com/dude/dms/startup/DirectoryManager.java
package com.dude.dms.startup;

import com.dude.dms.backend.brain.DmsLogger;
import com.dude.dms.backend.brain.FileManager;
import org.springframework.stereotype.Component;

import java.io.File;

import static com.dude.dms.backend.brain.OptionKey.DOC_POLL_PATH;

@Component
public class DirectoryManager {

    private static final DmsLogger LOGGER = DmsLogger.getLogger(DirectoryManager.class);

    public void checkDirectories() {
        File pollDir = new File(DOC_POLL_PATH.getString());
        if (!pollDir.exists()) {
            LOGGER.info("Creating input directory new docs {}", pollDir);
            pollDir.mkdir();
        }
        FileManager.createDirectories();
    }
}

File: src/main/kotlin/com/dude/dms/startup/DirectoryManager.kt
package com.dude.dms.startup

import com.dude.dms.backend.brain.DmsLogger
import com.dude.dms.backend.brain.FileManager
import com.dude.dms.backend.brain.OptionKey
import org.springframework.stereotype.Component
import java.io.File

@Component
class DirectoryManager {

    fun checkDirectories() {
        val pollDir = File(OptionKey.DOC_POLL_PATH.string)
        if (!pollDir.exists()) {
            LOGGER.info("Creating input directory new docs {}", pollDir)
            pollDir.mkdir()
        }
        FileManager.createDirectories()
    }

    companion object {
        private val LOGGER = DmsLogger.getLogger(DirectoryManager::class.java)
    }
}